### PR TITLE
🛑 Global useIsWrapping hook

### PR DESCRIPTION
**Pull request by W3stside**
Part of #464 

Global `useIsWrapping` hook to keep swap direct wrap and ethwethswap `button` state in sync

When ETH selling and wrapping then switching to ETH/WETH direct wrap WHILE a wrapping tx is pending:

First:
Screenshot from 2021-04-20 15-46-04

Immediately switch to WETH as buy while pending tx:
Screenshot from 2021-04-20 15-46-14

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]

File: src/custom/hooks/useIsWrapping.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const transactions = useAllTransactions()
+
+  if (!chainId || !transactions) return false
+
+  const txValues: TransactionDetails[] = Object.values(transactions)
+
+  // filter for any tx with Wrap in the summary
+  // then at least 1 without a receipt (pending) that is recent
+  return txValues
+    .filter(({ summary }) => summary?.includes('Wrap'))
```
Comment (anxolin) [inline src/custom/hooks/useIsWrapping.ts:16]: I'm ok with that. It just feels a bit weird to not have types in the transaction, so we can differentiate them.

File: src/custom/hooks/useIsWrapping.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    .some((tx: TransactionDetails) => !tx.receipt && isTransactionRecent(tx))
```
Comment (anxolin) [inline src/custom/hooks/useIsWrapping.ts:17]: I'm starting to think. Will we have an issue with tx that we speed up?

**Review by anxolin — COMMENTED** [review]
Great PR! not approving yet, because I want to double check if we have an issue with speeded up tx...

**Comment by W3stside** [comment]
sped up tx in MM does nothing for `Pending Transactions` thus nothing for this either. it would eventually pass after the `isTransactionRecent` fn validation failed

**Comment by anxolin** [comment]
We can look at this together with this one https://github.com/gnosis/gp-swap-ui/issues/212

**Pull request closed without merging**